File: JAVA/java/src/FindtheWinneroftheCircularGame.java
public class FindtheWinneroftheCircularGame {
    public static int findTheWinner(int n, int k) {
        int ans = 0;
        for (int i = 2; i <= n; i++) {
            ans = (ans + k) % i;
        }
        // add 1 to convert back to 1 indexing
        return ans + 1;
    }

    public static void main(String[] args) {
        System.out.println(findTheWinner(5,2));
    }
}

File: python/Find the Winner of the Circular Game.py
def findTheWinner( n: int, k: int) -> int:
    ans = 0
    for i in range(2, n + 1):
        ans = (ans + k) % i
    # add 1 to convert back to 1 indexing
    return ans + 1
print(findTheWinner(6, 5))